### PR TITLE
add unsupported cxf module list

### LINKED ISSUE #857: document unsuported module
I propose to add to extensions list in doc, a list of unsupported module

### PR DESCRIPTION
**Pull request by dufoli**
fix #857 
add a documentation for unsupported module.

### CODE REVIEW

**Comment by dufoli** [comment]
@shumonsharif @ppalaga hello, I have just list all unsupported modules of CXF and alternative.

**Comment by ppalaga** [comment]
Welcome back @dufoli! Thanks for the list. Let me propose some spelling improvements.

File: docs/modules/ROOT/pages/reference/index.adoc [hunk 1]
```diff
@@ -56,9 +56,85 @@
 |Stable
 |1.5.11
 |
+
 // standards: END
 |===
 
+[[unsupported]]
+== Unsupported Modules
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/index.adoc:64]:
```suggestion
== Unsupported modules

Here is a list of CXF modules currently not supported by {quarkus-cxf-project-name} along with possible alternatives and/or reasons why the given module is not supported.
```

File: docs/modules/ROOT/pages/reference/index.adoc [hunk 2]
```diff
@@ -56,9 +56,85 @@
...
+
+|===
+| Quarkus CXF extension | Alternative
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/index.adoc:67]:
```suggestion
| CXF module | Alternative
```

File: docs/modules/ROOT/pages/reference/index.adoc [hunk 3]
```diff
@@ -56,9 +56,85 @@
...
+
+| JAX-RS +
+`cxf-rt-frontend-jaxrs` `cxf-rt-rs-client`
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/index.adoc:70]:
```suggestion
`cxf-rt-frontend-jaxrs` +
`cxf-rt-rs-client`
```

File: docs/modules/ROOT/pages/reference/index.adoc [hunk 4]
```diff
@@ -56,9 +56,85 @@
...
+| https://quarkus.io/guides/rest-json[Use quarkus core rest-json support]
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/index.adoc:71]:
```suggestion
| https://quarkus.io/guides/rest-json[Use Quarkus RESTEasy] 
```

File: docs/modules/ROOT/pages/reference/index.adoc [hunk 5]
```diff
@@ -56,9 +56,85 @@
...
+
+|===
+| Quarkus CXF extension | Alternative
+
+| JAX-RS +
+`cxf-rt-frontend-jaxrs` `cxf-rt-rs-client`
+| https://quarkus.io/guides/rest-json[Use quarkus core rest-json support] 
+
+| https://cxf.apache.org/fediz.html[Fediz]
+| legacy SAML protocol is not supported but openId connect is supported on Quarkus Core
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/index.adoc:74]:
```suggestion
| Use Quarkus OpenID Connect
```

Which link could we put here? Is https://quarkus.io/guides/security-oidc-bearer-token-authentication-tutorial sufficient? Or is this also meant for client?

File: docs/modules/ROOT/pages/reference/index.adoc [hunk 6]
```diff
@@ -56,9 +56,85 @@
...
+
+| https://cxf.apache.org/docs/aegis-21.html[Aegis] 
+| use JAXB or JAXWS
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/index.adoc:77]:
```suggestion
| Use JAXB and JAX-WS
```

File: docs/modules/ROOT/pages/reference/index.adoc [hunk 7]
```diff
@@ -56,9 +56,85 @@
...
+
+| https://cxf.apache.org/dosgi-architecture.html[DOSGI] +
+Karaf
+|none
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/index.adoc:81]:
```suggestion
Karaf
| `quarkus-cxf`
```

File: docs/modules/ROOT/pages/reference/index.adoc [hunk 8]
```diff
@@ -56,9 +56,85 @@
...
+
+| Jibx 
+| Use JAXB or JAXWS
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/index.adoc:84]:
```suggestion
| JiBX 
| Use JAXB and JAX-WS
```

File: docs/modules/ROOT/pages/reference/index.adoc [hunk 9]
```diff
@@ -56,9 +56,85 @@
...
+
+| Local transport +
+`cxf-rt-transports-local`
+| use regular http transport
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/index.adoc:88]:
```suggestion
| Use HTTP transport
```

File: docs/modules/ROOT/pages/reference/index.adoc [hunk 10]
```diff
@@ -56,9 +56,85 @@
...
+
+| JMS transport +
+`cxf-rt-transports-jms`
+| use regular http transport
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/index.adoc:92]:
```suggestion
| Use HTTP transport
```

File: docs/modules/ROOT/pages/reference/index.adoc [hunk 11]
```diff
@@ -56,9 +56,85 @@
...
+
+| JBI +
+`cxf-rt-transports-jbi` `cxf-rt-bindings-jbi`
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/index.adoc:95]:
```suggestion
`cxf-rt-transports-jbi` +
`cxf-rt-bindings-jbi`
```

File: docs/modules/ROOT/pages/reference/index.adoc [hunk 12]
```diff
@@ -56,9 +56,85 @@
...
+| deprecated on CXF use regular http transport
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/index.adoc:96]:
```suggestion
| Deprecated in CXF use HTTP transport
```

File: docs/modules/ROOT/pages/reference/index.adoc [hunk 13]
```diff
@@ -56,9 +56,85 @@
...
+
+| UDP transport +
+`cxf-rt-transports-udp`
+| use regular http transport
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/index.adoc:100]:
```suggestion
| Use HTTP transport
```

File: docs/modules/ROOT/pages/reference/index.adoc [hunk 14]
```diff
@@ -56,9 +56,85 @@
...
+
+| https://cxf.apache.org/docs/coloc-feature.html[Coloc transport]
+| use regular http transport
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/index.adoc:103]:
```suggestion
| Use HTTP transport
```

File: docs/modules/ROOT/pages/reference/index.adoc [hunk 15]
```diff
@@ -56,9 +56,85 @@
...
+
+| https://cxf.apache.org/docs/websocket.html[Websocket transport] +
+`cxf-rt-transports-websocket`
+| use regular http transport
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/index.adoc:107]:
```suggestion
| https://cxf.apache.org/docs/websocket.html[WebSocket transport] +
`cxf-rt-transports-websocket`
| Use HTTP transport
```

File: docs/modules/ROOT/pages/reference/index.adoc [hunk 16]
```diff
@@ -56,9 +56,85 @@
...
+
+| Clustering +
+`cxf-rt-features-clustering`
+| None (but plan to be supported)
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/index.adoc:111]:
```suggestion
| [Planned](https://github.com/quarkiverse/quarkus-cxf/issues/836)
```

File: docs/modules/ROOT/pages/reference/index.adoc [hunk 17]
```diff
@@ -56,9 +56,85 @@
...
+
+| CORBA
+`cxf-rt-bindings-corba`
+| deprecated use JAXB or JAXWS
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/index.adoc:115]:
```suggestion
| Deprecated in CXF, use JAX-WS
```

File: docs/modules/ROOT/pages/reference/index.adoc [hunk 18]
```diff
@@ -56,9 +56,85 @@
...
+
+| https://cxf.apache.org/docs/sdo.html[SDO databinding] +
+`cxf-rt-databinding-sdo`
+| 
+
+| https://cxf.apache.org/docs/xmlbeans.html[XMLBeans]
+| deprecated on CXF
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/index.adoc:122]:
```suggestion
| Deprecated on CXF
```

File: docs/modules/ROOT/pages/reference/index.adoc [hunk 19]
```diff
@@ -56,9 +56,85 @@
...
+
+| https://cxf.apache.org/docs/javascript.html[Javascript frontend]
+| Use JAXB or JAXWS
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/index.adoc:125]:
```suggestion
| Use JAX-WS 
```

File: docs/modules/ROOT/pages/reference/index.adoc [hunk 20]
```diff
@@ -56,9 +56,85 @@
...
+| 
+
+| https://cxf.apache.org/docs/xmlbeans.html[XMLBeans]
+| deprecated on CXF
+
+| https://cxf.apache.org/docs/javascript.html[Javascript frontend]
+| Use JAXB or JAXWS 
+
+| https://cxf.apache.org/docs/using-cxf-jca-rar-in-application-server.html[JCA transport]
+| use regular http transport
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/index.adoc:128]:
```suggestion
| Use HTTP transport
```

File: docs/modules/ROOT/pages/reference/index.adoc [hunk 21]
```diff
@@ -56,9 +56,85 @@
...
+
+| WS-Transfer runtime
+| None
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/index.adoc:131]:
```suggestion
| WS-Transfer runtime +
`cxf-rt-ws-transfer`
|
```

File: docs/modules/ROOT/pages/reference/index.adoc [hunk 22]
```diff
@@ -56,9 +56,85 @@
...
+
+| throttling
+| None
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/index.adoc:134]:
```suggestion
| Throttling +
`cxf-rt-features-throttling`
|
```

**Review by ppalaga — COMMENTED** [review]
Some suggestions inline

**Comment by ppalaga** [comment]
@dufoli if you rebase on top of recent `main` the CI failure should go away.

File: docs/modules/ROOT/pages/reference/index.adoc (hunk 5 above)
Comment (dufoli) [reply docs/modules/ROOT/pages/reference/index.adoc:77]: best is https://quarkus.io/guides/security-oidc-code-flow-authentication-tutorial I think

File: docs/modules/ROOT/pages/reference/index.adoc (hunk 7 above)
Comment (dufoli) [reply docs/modules/ROOT/pages/reference/index.adoc:84]: for this one I do not understand the change.

Comment (ppalaga) [reply docs/modules/ROOT/pages/reference/index.adoc]: I thought Karaf was an OSGi based container for web applications. So Quarkus and specifically Quarkus CXF could be seen as an alternative, no?

**Comment by dufoli** [comment]
Indeed it was a first shoot because the big job was to list all module of cxf on maven central and on documentation and check what is supported or not.

File: docs/modules/ROOT/pages/reference/index.adoc [hunk 23]
```diff
@@ -56,9 +56,91 @@
 |Stable
 |1.5.11
 |
+
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/index.adoc:59]:
```suggestion
```

Could you please remove this change? The Part between standards: START and standards: END is generated at build time, so it gets removed and therefore the CI complains about unexpected changes in the source tree after the build. 

Running `mvn -f docs/ -am generate-resources` would do the same.

**Comment by ppalaga** [comment]
Thanks a lot, @dufoli !

**Pull request merged**